### ISSUE TITLE
False positive for "Style/HashSyntax" when using a variable as a hash key

### ISSUE THREAD

**Issue by mehanoid** [rubocop/rubocop#10401]
If you use the same variable as both a hash key and its value, then ruboсop wants to omit the value. The proposed autocorrection breaks the semantics of the code.

## Expected behavior
No offences for the following code:
```ruby
# frozen_string_literal: true

var = 1
{ var => var } # resulting in { 1 => 1}
```

## Actual behavior
```
For /home/meh/Projects/rubocop-test: Default configuration from /home/meh/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/bundler/gems/rubocop-8ae9e0f2081d/config/default.yml
Inspecting 1 file
Scanning /home/meh/Projects/rubocop-test/hash.rb
Loading cache from /home/meh/.cache/rubocop_cache/da7c0adf3cdca36c6db455c7f909e01abe58503a/0536d18c9eb040c724be8df71fd74a67730b58ef/027b8e17147ee3c7f4fce30a50ffb44ff511b6cb
C

Offenses:

hash.rb:4:10: C: [Corrected] Style/HashSyntax: Omit the hash value.
{ var => var }
         ^^^

1 file inspected, 1 offense detected, 1 offense corrected
Finished in 0.18334346099800314 seconds
```

The code after autocorrection:
```ruby
# frozen_string_literal: true

var = 1
{ var: } # resulting in { var: 1}
```

## Steps to reproduce the problem
See above

## RuboCop version
```
❯ rubocop -V
1.25.1 (using Parser 3.1.0.0, rubocop-ast 1.15.1, running on ruby 3.1.0 x86_64-linux)
```

**Comment by koic** [comment]
This is not a false positive because it is an expected behavior of `EnforcedShorthandSyntax: always` with Ruby 3.1. If you would not like to omit the hash value, please set `EnforcedShorthandSyntax: never` or `EnforcedShorthandSyntax: either`.
https://docs.rubocop.org/rubocop/1.25/cops_style.html#enforcedshorthandsyntax-always-default

**Comment by mehanoid** [comment]
@koic hash value omission is only possible then key is a Symbol, otherwise it just breaks the code like in the example above.

**Comment by mehanoid** [comment]
I've updated the examples with comments reflecting the result of running the code

**Comment by koic** [comment]
Ah, you are right. It's my oversight.
